src/lib.rs

```diff
@@ -1044,8 +1048,9 @@ impl<L: SynthLanguage> Synthesizer<L> {
         runner.egraph.rebuild();
         self.egraph = runner.egraph;
 
-        assert!(self.params.iters > 0);
-        for iter in 1..=self.params.iters {
+        let iters = self.params.iters.expect("iters required for rule lifting");
```

Comment:
this is more a question for @bksaiki: didn't you say you were using custom workloads for trig? How does that work with iters?